editor/src/components/canvas/ui-jsx-canvas.tsx

```diff
@@ -775,3 +793,16 @@ const CanvasContainer = React.forwardRef<
   )
 })
 CanvasContainer.displayName = 'CanvasContainer'
+
+function pathArraysEqual(one: Array<ElementPath>, other: Array<ElementPath>) {
+  if (one.length != other.length) {
+    return false
+  }
+
+  for (const path of one) {
```

Comment:
There is a potential bug here as we're not enforcing uniqueness on either of the arrays, so e.g. if we called this with `pathArraysEqual([path1, path1], [path1, path2])` it would return. In this specific use case that would be fine, since we only care that all of the values of `ElementsToRerenderGLOBAL.current` are contained in `elementsToRerenderRef.current`, and not that they contain the exact same values (I believe?), so I think it's not necessarily a problem to leave it as it is (so that we don't have to enforce uniqueness here), but I'd suggest we call that out in a comment.